### PR TITLE
Fix gradients of sinh and cosh

### PR DESCRIPTION
**Pull request by gbaydin**
Hi there, thank you for the awesome work you've been doing on this!

I noticed that the code does not properly propagate gradients through ```sinh``` and ```cosh``` nodes. I'm submitting a little fix for this.

### CODE REVIEW

**Comment by alexbw** [comment]
Ah, nice catch! Would you mind adding a test for these functions (and any other ones you can think of) so that this doesn't regress, and then I'll merge?
You might take [inspiration from the cmax/cmin gradient test](https://github.com/twitter/torch-autograd/blob/dae7f5393101f8806392cc698e7656a0a873250f/test/test.lua#L596-L612) to easily test a lot of different functions.

**Comment by gbaydin** [comment]
Sure, let me have a look at it.

**Comment by gbaydin** [comment]
I wrote some basic tests for ```sinh```, ```cosh```, and ```tanh```. The tests successfully detect the problem before the fix. Please note that the multiplication in

```lua
return 2 * torch.sum(torch[f](params.x))
```

is crucial for catching the error.

There are two issues:

* For some reason, the test for ```Optimized_LessThan``` is failing on Travis CI. I believe that this is unrelated to the changes in this pull request.

* I also looked into adding some more tests for other unary functions. Unfortunately there is a problem with the testing code (```gradcheck.lua```) that could be fixed before this, so I would like to leave it for another pull request. Briefly, the problem is that with ```randomizeInput``` some functions (such as ```log``` or ```sqrt```) can generate gradients with one or more ```nan``` elements. When this happens, the comparison between ```grads``` and ```gradsVerify``` in ```jacobianFromAutograd``` fails, and we get  "autograd gradient not deterministic".

**Comment by alexbw** [comment]
The LessThan gradient seems to fail probabilistically, I need to sort that out soon.
Thank you for reporting the problem with nans, I'll look into that as well.

**Comment by alexbw** [comment]
Awesome, thanks for the contribution!

**Comment by alexbw** [comment]
@gbaydin The Sinh test is failing, would you mind having a look?
